content/docs/studio/rest-api.md

```diff
@@ -0,0 +1,58 @@
+# REST API
+
+Studio REST API is WIP and will be extended. The API is hosted under the `/api`
+route on the Studio server: https://studio.iterative.ai/api or
+https://your-domain/api in case of
+[self-hosted Studio](/doc/studio/self-hosting/installation).
+
+To use API, you need to generate
+[Studio access token](/doc/studio/user-guide/account-management#studio-access-token).
+
+## Download model
+
+Get signed url to download the model binary. Requires the model to be stored
+with DVC with s3 or azure remote. `path` in model annotation must point to a
+single file and not to a directory. Note, that you need to
+[set up remote cloud credentials](/doc/studio/user-guide/account-management#cloud-credentials)
+for Studio have rights to sign urls.
+
+```yaml
+Endpoint: api/model-registry/get-download-uris
+HTTP Method: GET
+```
+
+### Request
+
+```yaml
+params:
+  repo:
+    desc: Git repo URL
+    type: string
+    required: true
+  name:
+    desc: Model name
+    type: string
+    required: true
+  version:
+    desc: Model version
+    type: string
+    required: true
+headers:
+  Authorization: 'token <TOKEN>'
+```
+
+### Response
+
+Response is a json-encoded dict. If the request was successful, keys will be
+paths to files inside the repo, and values will be signed urls you can query to
+actually download the model.
+
+### Example curl
+
+```
+$ curl https://studio.iterative.ai/api/model-registry/get-download-uris?repo=git@github.com:iterative/demo-bank-customer-churn.git&name=randomforest-model&version=v2.0.0 --header "Authorization:token <TOKEN>"
+
+{
```

Comment:
I guess it's to handle directories and keep the info about the directory structure. Agree it complicates the happy path. Are there other options? Could we make a directory return a url that has an index page to all the individual file paths in it (something like [this](https://stackoverflow.com/questions/273743/using-wget-to-recursively-fetch-a-directory-with-arbitrary-files-in-it))?

Reply:
Right - this is because of directories. Interesting solution with wget - didn't think of this. Following that page, looks like we need to host html page that a user can run `wget` against. And API should return that page URL. I wonder, is there a simpler way, like just returning actual html page in response and running `wget` against it?